src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -94,12 +96,113 @@ void EventPipe::Initialize()
 #endif
     }
 
-
     {
         CrstHolder _crst(GetLock());
         if (tracingInitialized)
             s_state = EventPipeState::Initialized;
     }
+#ifdef FEATURE_EVENTPIPE_STARTUP
+    EnableViaEnvironmentVariables();
+#endif // FEATURE_EVENTPIPE_STARTUP
+}
+
+//
+// If EventPipe environment variables are specified, parse them and start a session
+// 
+void EventPipe::EnableViaEnvironmentVariables()
+{
+    STANDARD_VM_CONTRACT;
+    if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EnableEventPipe) != 0)
+    {
+        LPWSTR eventpipeConfig = NULL;
+        CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeConfig, &eventpipeConfig);
+        LPCWSTR eventpipeOutputPath = Configuration::GetKnobStringValue(W("EventPipeOutputPath"));
+        uint32_t eventpipeCircularBufferMB = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeCircularMB);
+
+        if (eventpipeOutputPath == NULL)
+        {
+            eventpipeOutputPath = W("trace.nettrace");
+        }
+        auto configuration = XplatEventLoggerConfiguration();
+        LPWSTR configToParse = eventpipeConfig;
+        int providerCnt = 0;
+
+        // Create EventPipeProviderConfiguration and start tracing.
+        EventPipeProviderConfiguration* pProviders = nullptr;
+
+        // If COMPlus_EnableEventPipe is set to 1 but no configuration was specified, enable EventPipe session
+        // with the default provider configurations.
+        if (configToParse == nullptr || *configToParse == L'\0')
+        {
+            providerCnt = 2;
+            pProviders = new EventPipeProviderConfiguration[providerCnt];
+            pProviders[0] = EventPipeProviderConfiguration(W("Microsoft-Windows-DotNETRuntime"), 0x4c14fccbd, 5, nullptr);
+            pProviders[1] = EventPipeProviderConfiguration(W("Microsoft-Windows-DotNETRuntimePrivate"), 0x4002000b, 5, nullptr);
+        }
+        else
+        {
+            // Count how many providers there are to parse
+            static WCHAR comma = W(',');
+            while(configToParse != nullptr)
+            {
+                providerCnt += 1;
+                auto end = wcschr(configToParse, comma);
+                if (end == nullptr)
+                {
+                    break;
+                }
+                configToParse = end + 1;
+            }
+            configToParse = eventpipeConfig;
+            pProviders = new EventPipeProviderConfiguration[providerCnt];
+            int i = 0;
+            while (configToParse != nullptr)
```

Comment:
This loop condition seems suspicious.  Apart from `eventpipeConfig` being `nullptr`, there's no other way that it could be `nullptr` here.  Maybe you wanted `*configToParse != '\0'`?